### PR TITLE
NWM_UDS:: Allow multiple BindNodes per channel

### PR DESCRIPTION
**Pull request by B3n30**
This should fix multiplayer in Code of Princess. The only game I know of that uses multiple binds per channel. I wasn't able to test it, since I don't own the game.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3519)

### CODE REVIEW

File: src/core/hle/service/nwm/nwm_uds.cpp [hunk 1]
```diff
@@ -692,9 +693,8 @@ void NWM_UDS::Bind(Kernel::HLERequestContext& ctx) {
                                        "NWM::BindNodeEvent" + std::to_string(bind_node_id));
     std::lock_guard<std::mutex> lock(connection_status_mutex);
 
-    ASSERT(channel_data.find(data_channel) == channel_data.end());
-    // TODO(B3N30): Support more than one bind node per channel.
-    channel_data[data_channel] = {bind_node_id, data_channel, network_node_id, event};
+    ASSERT(bind_node_data.find(bind_node_id) == bind_node_data.end());
```
Comment (wwylele) [inline src/core/hle/service/nwm/nwm_uds.cpp]: What does hardware do if the nod is already bound?

File: src/core/hle/service/nwm/nwm_uds.cpp [hunk 2]
```diff
@@ -318,21 +318,22 @@ static void HandleSecureDataPacket(const Network::WifiPacket& packet) {
...
-    if (channel_info == channel_data.end())
-        return;
 
-    if (channel_info->second.network_node_id != BroadcastNetworkNodeId &&
-        channel_info->second.network_node_id != secure_data.src_node_id)
-        return;
+    for (auto bind_node : bind_node_data) {
+        // Ignore packets from channels we're not interested in.
+        if (bind_node.second.channel != secure_data.data_channel)
+            break;
```
Comment (wwylele) [inline src/core/hle/service/nwm/nwm_uds.cpp]: Shouldn't this continue to check the next element in bind_node_data?

File: src/core/hle/service/nwm/nwm_uds.cpp [hunk 3]
```diff
@@ -318,21 +318,22 @@ static void HandleSecureDataPacket(const Network::WifiPacket& packet) {
...
+
+        if (bind_node.second.network_node_id != BroadcastNetworkNodeId &&
+            bind_node.second.network_node_id != secure_data.src_node_id)
+            break;
```
Comment (wwylele) [inline src/core/hle/service/nwm/nwm_uds.cpp]: ditto

**Comment by wwylele** [comment]
Please fix the conflict, thanks!

**Comment by jroweboy** [comment]
This is a pretty simpleish change, and it would be great to have it. After you get some time to pick up development again, I'd love to see this get merged. I understand that you are busy though, so just reopen this when you get some more time.

**Comment by B3n30** [comment]
Rebased and fixed conflicts

**Comment by wwylele** [comment]
Got several reports saying this introduced connection issue, so please investigate them before merging. I think we once also put this in canary and had no problem? so might be rebasing error>

**Comment by B3n30** [comment]
Since I can't spot the mistake that causes the issue in that PR I will untag it from canary...

**Comment by B3n30** [comment]
I'm closing this since I can't find the mistake that causes the error. I will probably reopen an issue for that instead

**Pull request closed without merging**